pkg/controller/infrastructure/actuator_delete.go

```diff
@@ -17,8 +22,35 @@ import (
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 )
 
-func (a *actuator) delete(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure, cluster *extensionscontroller.Cluster) error {
-	infrastructureConfig, infrastructureStatus, err := a.decodeInfrastructure(infrastructure)
+type firewallDeleter struct {
+	ctx                  context.Context
+	logger               logr.Logger
+	c                    client.Client
+	infrastructure       *extensionsv1alpha1.Infrastructure
+	infrastructureConfig *metalapi.InfrastructureConfig
+	providerStatus       *metalapi.InfrastructureStatus
+	cluster              *extensionscontroller.Cluster
+	mclient              *metalgo.Driver
+	clusterID            string
+	clusterTag           string
+	machineID            string
+	projectID            string
+}
+
+func (a *actuator) Delete(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure, cluster *extensionscontroller.Cluster) error {
+	return Delete(ctx, a.logger, a.restConfig, a.client, a.decoder, infrastructure, cluster)
+}
+
+func Delete(
```

Comment:
Please dont use ancient C-Style function signatures.

Reply:
That's the only place where I left that in because I did want to have a similar-looking interface to how SAP does it in their controllers: https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/controller/infrastructure/actuator_reconcile.go#L41-L47

After this call, I wrap everything into a helper struct.

Reply:
Okay, I removed that.